gen/xbuild.sh

```diff
@@ -19,69 +19,69 @@ mkdir dist || exit 5
 ERRORS=
 
 build() {
-    echo "=== Building for GOOS=$1 and GOARCH=$2."
     # https://golang.org/doc/install/source#environment
+    echo "=== Building for GOOS=$1 and GOARCH=$2."
     CGO_ENABLED=0 GOOS="$1" GOARCH="$2" go build -o dist/ \
         -ldflags="-s -w -X main.gVersion=$version"
     if test "$?" != "0"; then
         ERRORS=1
     else
-        case "$3" in
-        *.tar.gz)
-            (
-                cd dist
-                tar czf "$3" "$4" --remove-files
-            ) || exit 5
-            ;;
-        *.zip)
-            (
-                cd dist
-                zip "$3" "$4" --move
-            ) || exit 5
-            ;;
-        esac
-        echo "dist/$3 successfully created."
+        (
+            cd dist || exit
+            OUTFILE=
+            case "$1" in
+            windows)
+                OUTFILE="lf-$1-$2.zip"
+                zip "$OUTFILE" lf.exe --move || exit
+                ;;
+            *)
+                OUTFILE="lf-$1-$2.tar.gz"
+                tar czf "$OUTFILE" lf --remove-files || exit
+                ;;
+            esac
+            echo "dist/$OUTFILE successfully created."
+        ) || exit 5
```

Comment:
My understanding is that the final return value is based on the last command, which is `echo "dist/$OUTFILE successfully created."`, so this will practically never happen unless `echo` somehow fails.

Reply:
It does happen because 

1) `exit` inside `()` only exits outside the `()` block, not the entire script
2) the plain `exit` inside the function returns the exit code of the last command. This is a bit cryptic, maybe I should change them.

Reply:
Perhaps this would be clearer if I made it a function...

Reply:
Ok, it should be clearer now.

Reply:
Ah OK, that makes sense, I thought `exit` by itself meant `exit 0`, thanks for letting me know.

Reply:
That's another thing I learned today... I think it's best not to use such cryptic features, though, so I changed it.